src/lib.rs

```diff
@@ -460,20 +460,20 @@ impl<const OSC_RAW_BUF_SIZE: usize> Parser<OSC_RAW_BUF_SIZE> {
                 c.len_utf8() - old_bytes
             },
             Err(err) => {
+                let valid_bytes = err.valid_up_to();
```

Comment:
I'd move the `if valid_bytes > 0` stuff here, should probably be the most hit branch anyway. Then either early return or else with a match that only has `to_copy` in none, doesn't really matter either way but  early return might look nicer wrt indentation.